src/Illuminate/Foundation/Configuration/ApplicationBuilder.php

```diff
@@ -148,16 +148,17 @@ public function withRouting(?Closure $using = null,
         ?string $pages = null,
         ?string $health = null,
         string $apiPrefix = 'api',
+        string $provider = AppRouteServiceProvider::class,
         ?callable $then = null)
     {
         if (is_null($using) && (is_string($web) || is_string($api) || is_string($pages) || is_string($health)) || is_callable($then)) {
             $using = $this->buildRoutingCallback($web, $api, $pages, $health, $apiPrefix, $then);
         }
 
-        AppRouteServiceProvider::loadRoutesUsing($using);
+        $provider::loadRoutesUsing($using);
```

Comment:
As the custom provider should be a subclass of the framework's `RouteServiceProvider`, we can keep the static call to the base provider here.

IMO, the `RouteServiceProvider@loadRoutesUsing` static method should not be overridden

Reply:
Your Point Is valid, but i think both will do the same